networkx/algorithms/similarity.py

```diff
@@ -1005,6 +1029,17 @@ def get_edit_paths(
     pending_u = list(G1.nodes)
     pending_v = list(G2.nodes)
 
+    initial_cost = 0
+    if roots:
+        assert len(roots) == 2, "Roots must be a 2-tuple."
+        root_u, root_v = roots
+        assert root_u in pending_u, "Root node not in graph."
+        assert root_v in pending_v, "Root node not in graph."
```

Comment:
```suggestion
        if len(roots) != 2:
            ValueError("roots must be a 2-tuple.")
        root_u, root_v = roots
        if (root_u not in pending_u) or (root_v not in pending_v):
            raise ValueError("root node not in graph.")
```
I think it's generally best-practice to raise specific exceptions rather than have assertions in the code - just a suggestion though! Perhaps there's a better nx-specific exception than ValueError for the second case?

Reply:
Thanks for this!  I was wondering about assert vs package specific Exception. 
There is a ```NetworkXNodeNotFound``` for the second case.

But should the first be a NetworkXError or a ValueError? What's best practice? We've put NetworkXError for most errors we flag.  If it is a ValueError couldn't we omit it and let the Python error raise on the next line?

Reply:
seems reasonable to me, probably `nx.NetworkXError` would be better, judging from @dschult 's previous commit.

Reply:
> But should the first be a NetworkXError or a ValueError? What's best practice? We've put NetworkXError for most errors we flag. If it is a ValueError couldn't we omit it and let the Python error raise on the next line?

Good point - Python will emit a ValueError on its own if the unpacking fails. The question becomes whether you want a specific message for the exception, i.e. `roots must be a 2-tuple` vs. `not enough values to unpack`. The most "Pythonic" approach might be:

```python
try:
    root_u, root_v = roots
except ValueError:
    raise ValueError('roots must be a 2-tuple')
```

but IMO any of the proposed solutions would be fine.

Reply:
I'm going to let the Python error stand for 'not enough values to unpack' since the code ```root_u, root_v = roots``` describes pretty well which input variable is causing the trouble.